src/textual/widgets/_tabbed_content.py

```diff
@@ -0,0 +1,148 @@
+from __future__ import annotations
+
+from itertools import zip_longest
+
+from rich.text import Text, TextType
+
+from ..app import ComposeResult
+from ..reactive import reactive
+from ..widget import Widget
+from ._content_switcher import ContentSwitcher
+from ._tabs import Tab, Tabs
+
+__all__ = [
+    "ContentTab",
+    "TabbedContent",
+    "TabPane",
+]
+
+
+class ContentTab(Tab):
+    """A Tab with an associated content id."""
+
+    def __init__(self, label: Text, content_id: str):
+        """Initialize a ContentTab.
+
+        Args:
+            label: The label to be displayed within the tab.
+            content_id: The id of the content associated with the tab.
+        """
+        super().__init__(label, id=content_id)
+
+
+class TabPane(Widget):
+    """A container for switchable content, with additional title.
+
+    This widget is intended to be used with [TabbedContent][textual.widgets.TabbedContent].
+
+    """
+
+    DEFAULT_CSS = """
+    TabPane {
+        height: auto;
+        padding: 1 2;
+    }
+    """
+
+    def __init__(
+        self,
+        title: TextType,
+        *children: Widget,
+        name: str | None = None,
+        id: str | None = None,
+        classes: str | None = None,
+        disabled: bool = False,
+    ):
+        """Initialize a TabPane.
+
+        Args:
+            title: Title of the TabPane (will be displayed in a tab label).
+            name: Optional name for the TabPane.
+            id: Optional ID for the TabPane.
+            classes: Optional initial classes for the widget.
+            disabled: Whether the TabPane is disabled or not.
+        """
+        self._title = self.render_str(title)
+        super().__init__(
+            *children, name=name, id=id, classes=classes, disabled=disabled
+        )
+
+
+class TabbedContent(Widget):
+    """A container with associated tabs to toggle content visibility."""
+
+    DEFAULT_CSS = """
+    TabbedContent {
+        height: auto;
+    }
+    TabbedContent > ContentSwitcher {
+        height: auto;
+    }
+    """
+
+    active: reactive[str] = reactive("", init=False)
+    """The ID of the active tab, or empty string if none are active."""
+
+    def __init__(self, *titles: TextType, initial: str = "") -> None:
+        """Initialize a TabbedContent widgets.
+
+        Args:
+            *titles: Positional argument will be used as title.
+            initial: The id of the initial tab, or empty string to select the first tab.
```

Comment:
As mentioned in a comment on the docs, but I felt it worth a mention any place this gets said, leaving this empty results in an error that `NoMatches: No child found with id=''`.